RubeesEat/Model/DB/DbCommandExtensions.cs

```diff
@@ -0,0 +1,36 @@
+using System.Data.Common;
+
+namespace RubeesEat.Model.DB;
+
+public static class DbCommandExtensions
+{
+
+    public static DbParameter AddParameter(this DbCommand command, string name, Guid value)
+    {
+        return AddParameterHelper(command, name, value);
+    }
+
+    public static DbParameter AddParameter(this DbCommand command, string name, decimal value)
+    {
+        return AddParameterHelper(command, name, value);
+    }
+
+    public static DbParameter AddParameter(this DbCommand command, string name, DateTime value)
+    {
+        return AddParameterHelper(command, name, value);
+    }
+
+    public static DbParameter AddParameter(this DbCommand command, string name, string value)
+    {
+        return AddParameterHelper(command, name, value);
+    }
+
+    private static DbParameter AddParameterHelper(this DbCommand command, string name, object value)
+    {
+        var parameter = command.CreateParameter();
+        parameter.ParameterName = name;
+        parameter.Value = value;
+        command.Parameters.Add(parameter);
+        return parameter;
+    }
```

Comment:
According to Patricks review, the benefit here is that only certain Datatypes are allowed for the parameter "value", (in our case only Guid, decimal, DateTime and string are allowed). This provides extra security. If you want, we can remove it, so we lose security but get a simpler code.

Reply:
Oh, but in this case the parameter never gets returned to the original caller, and we wanted to have them return the parameter, in case we need it. All the public AddParameter functions are void. The parameter that gets returned by the private helper function disappears because it is not used in a variable or returned further